src/Agent.Worker/TestResults/TestCaseResultDataConverter.cs

```diff
@@ -0,0 +1,76 @@
+            testCaseResultWebApi.ErrorMessage = testCaseResultData.ErrorMessage;
+            testCaseResultWebApi.FailureType = testCaseResultData.FailureType;
+            testCaseResultWebApi.Id = testCaseResultData.Id;
+            testCaseResultWebApi.IterationDetails = testCaseResultData.IterationDetails;
+            testCaseResultWebApi.LastUpdatedBy = testCaseResultData.LastUpdatedBy;
+            testCaseResultWebApi.LastUpdatedDate = testCaseResultData.LastUpdatedDate;
+            testCaseResultWebApi.Outcome = testCaseResultData.Outcome;
+            testCaseResultWebApi.Owner = testCaseResultData.Owner;
+            testCaseResultWebApi.Priority = testCaseResultData.Priority;
+            testCaseResultWebApi.Project = testCaseResultData.Project;
+            testCaseResultWebApi.Release = testCaseResultData.Release;
+            testCaseResultWebApi.ReleaseReference = testCaseResultData.ReleaseReference;
+            testCaseResultWebApi.ResetCount = testCaseResultData.ResetCount;
+            testCaseResultWebApi.ResolutionState = testCaseResultData.ResolutionState;
+            testCaseResultWebApi.ResolutionStateId = testCaseResultData.ResolutionStateId;
+            testCaseResultWebApi.ResultGroupType = testCaseResultData.ResultGroupType;
+            testCaseResultWebApi.Revision = testCaseResultData.Revision;
+            testCaseResultWebApi.RunBy = testCaseResultData.RunBy;
+            testCaseResultWebApi.StackTrace = testCaseResultData.StackTrace;
+            testCaseResultWebApi.StartedDate = testCaseResultData.StartedDate;
+            testCaseResultWebApi.State = testCaseResultData.State;
+            testCaseResultWebApi.TestCase = testCaseResultData.TestCase;
+            testCaseResultWebApi.TestCaseReferenceId = testCaseResultData.TestCaseReferenceId;
+            testCaseResultWebApi.TestCaseTitle = testCaseResultData.TestCaseTitle;
+            testCaseResultWebApi.TestPlan = testCaseResultData.TestPlan;
+            testCaseResultWebApi.TestPoint = testCaseResultData.TestPoint;
+            testCaseResultWebApi.TestRun = testCaseResultData.TestRun;
+            testCaseResultWebApi.TestSuite = testCaseResultData.TestSuite;
+            testCaseResultWebApi.Url = testCaseResultData.Url;
+
+            ConvertSubResults(testCaseResultData, testCaseResultWebApi);
+        }
+
+        private void ConvertSubResults(TestCaseResultData testCaseResultData, TestCaseResult testCaseResultWebApi)
+        {
+            if (testCaseResultData.TestCaseSubResultData == null || !testCaseResultData.TestCaseSubResultData.Any())
+            {
+                return;
+            }
+
+            testCaseResultWebApi.SubResults = new List<TestSubResult>();
+            foreach (var subResultData in testCaseResultData.TestCaseSubResultData)
+            {
+                var subResultWebApi = new TestSubResult();
+                new TestCaseSubResultDataConverter().Convert(subResultData, subResultWebApi);
```

Comment:
why not make all yuour convert class static? seems weird you have to new an object everytime.

Reply:
testCaseResultDataConverter and testCaseSubResultDataConverter are different classes.

Reply:
what about change both to Static? it's not a big deal, but just weird you have to create a new object everytime for convert.

Reply:
Done :)